tests/scancode/test_ignore_files.py

```diff
@@ -74,3 +81,45 @@ def test_ignore_glob_file(self):
             {}
         )
         assert is_ignored(*test)
+
+    def test_resource_paths_1(self):
+        test = (
+            test_dir,
+            {'*.doc': 'test ignore'}
+        )
+        expected = [
+            'user',
+            'user/src',
+            'user/src/test',
+            'user/src/test/sample.txt'
+        ]
+        for abs_path, rel_path in resource_paths(*test):
```

Comment:
> The generator object returned from the resource_paths object seems to have different order of elements on different operating systems.

Yes, the order is not stable. So you can just sort your results as here we do not care for ordering. And therefore make it a simple sorted list-based test